integration-tests/rpc_integration_test.go

```diff
@@ -0,0 +1,465 @@
+	assert.Equal(t, foundPeerLog.Protocol, protocolString)
+
+	// Cancel the context and wait for all outstanding goroutines to finish.
+	cancel()
+	wg.Wait()
+}
+
+func TestGetStats(t *testing.T) {
+	teardownSubTest := setupSubTest(t)
+	defer teardownSubTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Remove the old database and p2p files.
+	removeOldFiles(t, ctx)
+
+	buildStandaloneForTests(t, ctx)
+
+	logMessages := make(chan string, 1024)
+
+	count := int(atomic.AddInt32(nodeCount, 1))
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		startStandaloneNode(t, ctx, count, logMessages)
+	}()
+
+	log, err := waitForLogSubstring(ctx, logMessages, "started RPC server")
+	require.NoError(t, err, "RPC server didn't start")
+	var jsonLog map[string]interface{}
+	err = json.Unmarshal([]byte(log), &jsonLog)
+	require.NoError(t, err)
+
+	expectedGetStatsResponse := &rpc.GetStatsResponse{
+		Version:              "development",
+		PubSubTopic:          "/0x-orders/network/1337/version/1",
+		Rendezvous:           "/0x-mesh/network/1337/version/1",
+		PeerID:               jsonLog["myPeerID"].(string),
+		EthereumChainID:      1337,
+		LatestBlock:          rpc.LatestBlock{},
+		NumOrders:            0,
+		NumPeers:             0,
+		MaxExpirationTime:    "115792089237316195423570985008687907853269984665640564039457584007913129639935",
+		StartOfCurrentUTCDay: getUTCMidnightOfDate(time.Now()),
+	}
+
+	client, err := rpc.NewClient(standaloneRPCEndpoint + strconv.Itoa(rpcPort+count))
+	require.NoError(t, err)
+
+	getStatsResponse, err := client.GetStats()
+
+	// HACK(jalextowle): Zeroing the Number and Hash fields of LatestBlock
+	//                   allows us to test more of the "GetStats" response
+	//                   without being too restrictive about the blockchain
+	//                   that is being used.
+	getStatsResponse.LatestBlock = rpc.LatestBlock{}
+
+	require.NoError(t, err)
+	require.Equal(t, expectedGetStatsResponse, getStatsResponse)
+
+	// Cancel the context and wait for all outstanding goroutines to finish.
+	cancel()
+	wg.Wait()
+}
+
+func TestOrdersSubscription(t *testing.T) {
+	teardownSubTest := setupSubTest(t)
+	defer teardownSubTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Remove the old database and p2p files.
+	removeOldFiles(t, ctx)
+
+	buildStandaloneForTests(t, ctx)
+
+	logMessages := make(chan string, 1024)
+
+	count := int(atomic.AddInt32(nodeCount, 1))
+
+	// Set up the dummy handler with an addOrdersHandler
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		startStandaloneNode(t, ctx, count, logMessages)
+	}()
+
+	_, err := waitForLogSubstring(ctx, logMessages, "started RPC server")
+	require.NoError(t, err, "RPC server didn't start")
+
+	client, err := rpc.NewClient(standaloneRPCEndpoint + strconv.Itoa(rpcPort+count))
+	require.NoError(t, err)
+
+	orderEventChan := make(chan []*zeroex.OrderEvent)
```

Comment:
We should use a buffered channel for `orderEventChan`. As the documentation explains:

> Client buffers up to 8000 notifications before considering the subscriber dead. The subscription Err channel will receive ErrSubscriptionQueueOverflow. Use a sufficiently large buffer on the channel or ensure that the channel usually has at least one reader to prevent this issue.